ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/MigrationsUtil.java

```diff
@@ -111,6 +123,23 @@ static ClientOptions createClientOptions(
       options.setUseAlpn(useAlpn);
       options.setVerifyHost(verifyHost);
     }
+
+    if (requestHeaders != null) {
+      options.setRequestHeaders(requestHeaders);
+    }
+
     return options;
   }
+
+  private static Map<String, String> loadRequestHeaders(final String headersFile) {
+    if (headersFile == null || headersFile.trim().isEmpty()) {
+      return null;
```

Comment:
Maybe return an empty map here as well?

Reply:
I don't feel strongly about this one. It's an internal helper method and the rest of the code keeps null checks to be defensive anyway.